db/document.go

```diff
@@ -76,12 +76,60 @@ type syncData struct {
 	removedRevisionBodyKeys map[string]string // keys of non-winning revisions that have been removed (and so may require deletion), indexed by revID
 }
 
+func (doc *SyncData) HashRedact(salt string) SyncData {
+
+	syncData := SyncData{
+		CurrentRev:      doc.CurrentRev,
+		NewestRev:       doc.NewestRev,
+		Flags:           doc.Flags,
+		Sequence:        doc.Sequence,
+		UnusedSequences: doc.UnusedSequences,
+		RecentSequences: doc.RecentSequences,
+		History:         RevTree{},
+		Channels:        channels.ChannelMap{},
+		Access:          UserAccessMap{},
+		RoleAccess:      UserAccessMap{},
+		Expiry:          doc.Expiry,
+		Cas:             doc.Cas,
+		Crc32c:          doc.Crc32c,
+		TombstonedAt:    doc.TombstonedAt,
+		Attachments:     doc.Attachments,
```

Comment:
Attachment names are probably considered UserData actually after reviewing this list.